File: fathersDay/Dsa/Day 26 (700. Search in a Binary Search Tree)/Solution.cpp
#include <iostream>

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
};

class Solution {
public:
    TreeNode* SearchBST(TreeNode* root, int val) {
        if (!root || root->val == val) {
            return root;
        } else if (val < root->val) {
            return SearchBST(root->left, val);
        } else {
            return SearchBST(root->right, val);
        }
    }
};

int main() {
    Solution solution;
    TreeNode* root = new TreeNode(4);
    root->left = new TreeNode(2);
    root->right = new TreeNode(7);
    root->left->left = new TreeNode(1);
    root->left->right = new TreeNode(3);
    int val = 2;
    TreeNode* result = solution.SearchBST(root, val);
    if (result) {
        std::cout << result->val << std::endl; // Output: 2
    } else {
        std::cout << "Value not found in the BST" << std::endl;
    }
    return 0;
}

File: fathersDay/Dsa/Day 26 (700. Search in a Binary Search Tree)/Solution.java
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode(int val) { this.val = val; }
}

class Solution {
    public TreeNode SearchBST(TreeNode root, int val) {
        if (root == null || root.val == val) {
            return root;
        } else if (val < root.val) {
            return SearchBST(root.left, val);
        } else {
            return SearchBST(root.right, val);
        }
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        TreeNode root = new TreeNode(4);
        root.left = new TreeNode(2);
        root.right = new TreeNode(7);
        root.left.left = new TreeNode(1);
        root.left.right = new TreeNode(3);
        int val = 2;
        TreeNode result = solution.SearchBST(root, val);
        if (result != null) {
            System.out.println(result.val); // Output: 2
        } else {
            System.out.println("Value not found in the BST");
        }
    }
}
